internal/overlord/servstate/handlers.go

```diff
@@ -23,6 +23,13 @@ import (
 	"github.com/canonical/pebble/internal/strutil/shlex"
 )
 
+func init() {
+	// Reap sub-children via setting the child subreaper
+	// that way we can reap "grandchildren" from our main process,
+	// effectively we can wait for the children of the terminating process
+	unix.Prctl(unix.PR_SET_CHILD_SUBREAPER, 1, 0, 0, 0)
+}
+
```

Comment:
I think setting ourselves as a subreaper is the right thing for Pebble -- good find.

However, let's not do it in a package-level `init()`. That means importing the package has significant side effects. I think we should just do this in `servstate.NewManager` -- that way we can check and return an error too.

Reply:
I think we can also clarify the comment a bit. How about something like this?

```
// Set Pebble as a "child subreaper" so it becomes the parent of orphaned children
// rather than PID 1, allowing us to wait for processes that a service starts (see
// https://unix.stackexchange.com/a/250156/73491).
```

Reply:
Makes sense, I saw some `init()` funcs in the code and wasn't sure where this piece belongs best, agree on not having side effects and thanks for the pointer to `NewManager`, I've added it there now.
Comment is great and I just set it as is.